addons/l10n_it_edi/models/account_move.py

```diff
@@ -0,0 +1,1212 @@
+                unsent_move.l10n_it_edi_header = text_message
+                unsent_move.sudo().message_post(body=html_message)
+                messages_to_log.append(text_message)
+            raise UserError("\n".join(messages_to_log)) from e
+
+        # Handle results
+        for filename, vals in results.items():
+            sent_move = self.filtered(lambda m: m.l10n_it_edi_attachment_id.name == filename)
+            if 'error' in vals and vals.get('error_level', '') == 'error':
+                sent_move.l10n_it_edi_state = False
+                sent_move.l10n_it_edi_transaction = False
+                message = nl2br(escape(_("Error uploading the e-invoice file %s.\n%s", filename, vals['error'])))
+            else:
+                is_demo = vals['id_transaction'] == 'demo'
+                sent_move.l10n_it_edi_state = 'processing'
+                sent_move.l10n_it_edi_transaction = vals['id_transaction']
+                message = (
+                    _("We are simulating the sending of the e-invoice file %s, as we are in demo mode.", filename)
+                    if is_demo else _("The e-invoice file %s was sent to the SdI for processing.", filename))
+            sent_move.l10n_it_edi_header = message
+            sent_move.sudo().message_post(body=message)
+
+    # -------------------------------------------------------------------------
+    # Update notifications
+    # -------------------------------------------------------------------------
+
+    def _l10n_it_edi_demo_mode_update_send_state(self):
+        for move in self:
+            filename = move.l10n_it_edi_attachment_id and move.l10n_it_edi_attachment_id.name or '???'
+            self._l10n_it_edi_write_send_state(
+                transformed_notification={
+                    'l10n_it_edi_state': 'forwarded',
+                    'l10n_it_edi_transaction': f'demo_{uuid.uuid4()}',
+                    'send_ack_to_edi_proxy': False,
+                    'date': fields.Date.today(),
+                    'filename': filename},
+                message=_("The e-invoice file %s has been sent in Demo EDI mode.", filename))
+
+    def _l10n_it_edi_update_send_state(self, notifications):
+        acks = {'transaction_ids': [], 'states': []}
+        for move in self:
+            notification = notifications[move.l10n_it_edi_transaction]
+            parsed_notification = move._l10n_it_edi_parse_notification(notification)
+            transformed_notification = move._l10n_it_edi_transform_notification(parsed_notification)
+            message = move._l10n_it_edi_get_message(transformed_notification)
+            move._l10n_it_edi_write_send_state(transformed_notification, message)
+            if (transformed_notification.get('send_ack_to_edi_proxy')
+                and (id_transaction_to_ack := transformed_notification.get('l10n_it_edi_transaction'))
+                and (ack_state := transformed_notification.get('l10n_it_edi_state'))):
```

Comment:
```py
if (
    transformed_notification.get('send_ack_to_edi_proxy')
    and (id_transaction_to_ack := transformed_notification.get('l10n_it_edi_transaction'))
    and (ack_state := transformed_notification.get('l10n_it_edi_state'))
):
```
Otherwise, the code is hard to read because the indentation is unclear